libcperciva/network/network_connect.c

```diff
@@ -118,8 +119,13 @@ tryconnect(struct connect_cookie * C)
 	/* Try addresses until we find one which doesn't fail immediately. */
 	for (; C->sas[0] != NULL; C->sas++) {
 		/* Can we try to connect to this address? */
-		if ((C->s = sock_connect_nb(C->sas[0])) != -1)
-			break;
+		if (C->sa_b != NULL) {
+			if ((C->s = sock_connect_bind_nb(C->sas[0], C->sa_b)) != -1)
+				break;
+		} else {
+			if ((C->s = sock_connect_nb(C->sas[0])) != -1)
```

Comment:
Hmm, we could make this whole section:
```
if ((C->s = sock_connect_nb(C->sas[0], C->sa_b)) != -1)
        break;
```
(and rely on `sock_connect_nb` handling C->sa_b being NULL or not)

That said, it could be argued that it would be clearer to see the `_bind_nb` vs. `_nb`.  I'm not convinced by that argument, but it's @cperciva's call.